File: mobile/packages/expo-terminal-live-input/ios/ExpoTerminalLiveInputModule.swift
import ExpoModulesCore

public class ExpoTerminalLiveInputModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoTerminalLiveInput")

    View(TerminalLiveInputView.self) {
      Events(
        "onEditorStateTransaction",
        "onInputFocus",
        "onInputBlur",
        "onKeyPress",
        "onTerminalEnter"
      )

      Prop("editable") { (view: TerminalLiveInputView, editable: Bool) in
        view.setEditable(editable)
      }

      AsyncFunction("focusAsync") { view in
        view.focusField()
      }

      AsyncFunction("blurAsync") { view in
        view.blurField()
      }

      AsyncFunction("setTextAsync") { (view, text: String) in
        view.setText(text)
      }
    }
  }
}

File: mobile/packages/expo-terminal-live-input/android/src/main/java/expo/modules/terminalliveinput/ExpoTerminalLiveInputModule.kt
package expo.modules.terminalliveinput

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoTerminalLiveInputModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoTerminalLiveInput")

    View(TerminalLiveInputView::class) {
      Events(
        "onEditorStateTransaction",
        "onInputFocus",
        "onInputBlur",
        "onKeyPress",
        "onTerminalEnter"
      )

      Prop("editable") { view: TerminalLiveInputView, editable: Boolean ->
        view.setEditable(editable)
      }

      AsyncFunction("focusAsync") { view: TerminalLiveInputView ->
        return@AsyncFunction view.focusField()
      }

      AsyncFunction("blurAsync") { view: TerminalLiveInputView ->
        return@AsyncFunction view.blurField()
      }

      AsyncFunction("setTextAsync") { view: TerminalLiveInputView, text: String ->
        view.setText(text)
      }
    }
  }
}
